File: c++/Sorting and Searching/Apartments.cpp
#include<bits/stdc++.h>
using namespace std;

int main()
{
	//ios::sync_with_stdio(0);cin.tie(0);
	int n , m , k ;
	cin >> n >> m >> k ;
	vector<int> applicant(n)  , apartment(m);
	for (auto &u : applicant)
		cin >> u ;
	for (auto &u : apartment)
		cin >> u ;
	sort(applicant.begin(), applicant.end());
	sort(apartment.begin(), apartment.end());
	int a = 0 , b = 0 , ans = 0 ;
	while (a < n && b < m) {
		int diff = applicant[a] - apartment[b];
		if (abs(diff) <= k) {
			ans++;
			a++;
			b++;
		}
		else if (diff > 0)
			b++;
		else
			a++;
	}
	cout << ans << '\n';
}

File: python/sorting-and-searching/apartments.py
def allocate_apartments(desired_sizes: list[int], apartment_sizes: list[int], k: int):
    desired_sizes.sort()
    apartment_sizes.sort()

    i, j = 0, 0
    ans = 0

    while j < len(desired_sizes) and i < len(apartment_sizes):
        diff = apartment_sizes[i] - desired_sizes[j]
        if abs(diff) <= k:
            ans += 1
            j += 1
            i += 1

        elif diff < 0:
            i += 1

        else:
            j += 1
    return ans


def main():
    n, m, k = map(int, input().split())
    desired_sizes = list(map(int, input().split()))
    apartment_sizes = list(map(int, input().split()))
    print(allocate_apartments(desired_sizes, apartment_sizes, k))


main()
